zenodo/modules/deposit/utils.py

```diff
@@ -245,3 +247,33 @@ def suggest_language(q, limit=5):
     if lut:
         langs = ([lut, ] + [l for l in langs if l != lut])[:limit]
     return langs
+
+
+def is_user_verified(days=7):
+    """
+    Permission function that evaluates if the user can create a deposit.
+
+    This applies in addition to preexisting permissions.
+    """
+    from flask_security import current_user
+    if current_user.external_identifiers:
+        return True
+
+    error_message = (
+            'To create a deposit please verify your email.'
```

Comment:
Please verify your email address in order to upload. You can resend the verification email from [here](https://www.zenodo.org/account/settings/profile/). Alternatively, you can link your Zenodo account with either your GitHub or ORCID account.

Reply:
This error message is also generated as a REST response so instead of adding the link, I replaced it with the explanatory text.